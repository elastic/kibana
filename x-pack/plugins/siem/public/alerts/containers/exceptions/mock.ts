/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
import {
  CreateExceptionListItemSchemaPartial,
  CreateExceptionListSchemaPartial,
  ExceptionListItemSchema,
  ExceptionListSchema,
} from '../../../../../lists/common/schemas';

export const mockExceptionList: ExceptionListSchema = {
  id: '1',
  list_id: 'endpoint_list',
  _tags: ['endpoint', 'process', 'malware', 'os:linux'],
  tags: ['user added string for a tag', 'malware'],
  type: 'endpoint',
  description: 'This is a sample endpoint type exception',
  name: 'Sample Endpoint Exception List',
  created_at: '2020-04-23T00:19:13.289Z',
  created_by: 'user_name',
  tie_breaker_id: '77fd1909-6786-428a-a671-30229a719c1f',
  updated_at: '2020-04-23T00:19:13.289Z',
  updated_by: 'user_name',
  meta: {},
};

export const mockNewExceptionList: CreateExceptionListSchemaPartial = {
  list_id: 'endpoint_list',
  _tags: ['endpoint', 'process', 'malware', 'os:linux'],
  tags: ['user added string for a tag', 'malware'],
  type: 'endpoint',
  description: 'This is a sample endpoint type exception',
  name: 'Sample Endpoint Exception List',
};

export const mockNewExceptionItem: CreateExceptionListItemSchemaPartial = {
  list_id: 'endpoint_list',
  item_id: 'endpoint_list_item',
  _tags: ['endpoint', 'process', 'malware', 'os:linux'],
  tags: ['user added string for a tag', 'malware'],
  type: 'simple',
  description: 'This is a sample endpoint type exception',
  name: 'Sample Endpoint Exception List',
  entries: [
    {
      field: 'actingProcess.file.signer',
      operator: 'included',
      match: 'Elastic, N.V.',
      match_any: undefined,
    },
    {
      field: 'event.category',
      operator: 'included',
      match_any: ['process', 'malware'],
      match: undefined,
    },
  ],
};

export const mockExceptionItem: ExceptionListItemSchema = {
  id: '1',
  created_at: '2020-04-23T00:19:13.289Z',
  created_by: 'user_name',
  tie_breaker_id: '77fd1909-6786-428a-a671-30229a719c1f',
  updated_at: '2020-04-23T00:19:13.289Z',
  updated_by: 'user_name',
  list_id: 'endpoint_list',
  item_id: 'endpoint_list_item',
  comment: [],
  meta: {},
  _tags: ['endpoint', 'process', 'malware', 'os:linux'],
  tags: ['user added string for a tag', 'malware'],
  type: 'simple',
  description: 'This is a sample endpoint type exception',
  name: 'Sample Endpoint Exception List',
  entries: [
    {
      field: 'actingProcess.file.signer',
      operator: 'included',
      match: 'Elastic, N.V.',
      match_any: undefined,
    },
    {
      field: 'event.category',
      operator: 'included',
      match_any: ['process', 'malware'],
      match: undefined,
    },
  ],
};
