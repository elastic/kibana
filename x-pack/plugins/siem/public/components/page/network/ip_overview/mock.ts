/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { IpOverviewData } from '../../../../graphql/types';

export const mockData: Readonly<Record<string, IpOverviewData>> = {
  complete: {
    source: {
      firstSeen: '2019-02-07T17:19:41.636Z',
      lastSeen: '2019-02-07T17:19:41.636Z',
      autonomousSystem: { as_org: 'Test Org', asn: 'Test ASN', ip: '10.10.10.10' },
      geo: {
        continent_name: ['North America'],
        city_name: ['New York'],
        country_iso_code: ['US'],
        country_name: null,
        location: {
          lat: [40.7214],
          lon: [-74.0052],
        },
        region_iso_code: ['US-NY'],
        region_name: ['New York'],
      },
      host: {
        os: {
          kernel: ['4.14.50-v7+'],
          name: ['Raspbian GNU/Linux'],
          family: [''],
          version: ['9 (stretch)'],
          platform: ['raspbian'],
        },
        name: ['raspberrypi'],
        id: ['b19a781f683541a7a25ee345133aa399'],
        ip: ['10.10.10.10'],
        architecture: ['armv7l'],
      },
    },
    destination: {
      firstSeen: '2019-02-07T17:19:41.648Z',
      lastSeen: '2019-02-07T17:19:41.648Z',
      autonomousSystem: { as_org: 'Test Org', asn: 'Test ASN', ip: '10.10.10.10' },
      geo: {
        continent_name: ['North America'],
        city_name: ['New York'],
        country_iso_code: ['US'],
        country_name: null,
        location: {
          lat: [40.7214],
          lon: [-74.0052],
        },
        region_iso_code: ['US-NY'],
        region_name: ['New York'],
      },
      host: {
        os: {
          kernel: ['4.14.50-v7+'],
          name: ['Raspbian GNU/Linux'],
          family: [''],
          version: ['9 (stretch)'],
          platform: ['raspbian'],
        },
        name: ['raspberrypi'],
        id: ['b19a781f683541a7a25ee345133aa399'],
        ip: ['10.10.10.10'],
        architecture: ['armv7l'],
      },
    },
  },
};
