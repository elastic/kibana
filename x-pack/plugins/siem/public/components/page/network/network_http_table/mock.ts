/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { NetworkHttpData } from '../../../../graphql/types';

export const mockData: { NetworkHttp: NetworkHttpData } = {
  NetworkHttp: {
    edges: [
      {
        node: {
          _id: '/computeMetadata/v1/instance/virtual-clock/drift-token',
          domains: ['metadata.google.internal'],
          methods: ['get'],
          statuses: [],
          lastHost: 'suricata-iowa',
          lastSourceIp: '10.128.0.21',
          path: '/computeMetadata/v1/instance/virtual-clock/drift-token',
          requestCount: 1440,
        },
        cursor: {
          value: '/computeMetadata/v1/instance/virtual-clock/drift-token',
          tiebreaker: null,
        },
      },
      {
        node: {
          _id: '/computeMetadata/v1/',
          domains: ['metadata.google.internal'],
          methods: ['get'],
          statuses: ['200'],
          lastHost: 'suricata-iowa',
          lastSourceIp: '10.128.0.21',
          path: '/computeMetadata/v1/',
          requestCount: 1020,
        },
        cursor: {
          value: '/computeMetadata/v1/',
          tiebreaker: null,
        },
      },
      {
        node: {
          _id: '/computeMetadata/v1/instance/network-interfaces/',
          domains: ['metadata.google.internal'],
          methods: ['get'],
          statuses: [],
          lastHost: 'suricata-iowa',
          lastSourceIp: '10.128.0.21',
          path: '/computeMetadata/v1/instance/network-interfaces/',
          requestCount: 960,
        },
        cursor: {
          value: '/computeMetadata/v1/instance/network-interfaces/',
          tiebreaker: null,
        },
      },
      {
        node: {
          _id: '/downloads/ca_setup.exe',
          domains: ['www.oxid.it'],
          methods: ['get'],
          statuses: ['200'],
          lastHost: 'jessie',
          lastSourceIp: '10.0.2.15',
          path: '/downloads/ca_setup.exe',
          requestCount: 3,
        },
        cursor: {
          value: '/downloads/ca_setup.exe',
          tiebreaker: null,
        },
      },
    ],
    inspect: {
      dsl: [''],
      response: [''],
    },
    pageInfo: {
      activePage: 0,
      fakeTotalCount: 4,
      showMorePagesIndicator: false,
    },
    totalCount: 4,
  },
};
