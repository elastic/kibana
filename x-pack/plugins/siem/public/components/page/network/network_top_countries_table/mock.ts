/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { NetworkTopCountriesData } from '../../../../graphql/types';

export const mockData: { NetworkTopCountries: NetworkTopCountriesData } = {
  NetworkTopCountries: {
    totalCount: 524,
    edges: [
      {
        node: {
          source: {
            country: 'DE',
            destination_ips: 12,
            flows: 12345,
            source_ips: 55,
          },
          destination: null,
          network: {
            bytes_in: 3826633497,
            bytes_out: 1083495734,
          },
        },
        cursor: {
          value: '8.8.8.8',
        },
      },
      {
        node: {
          source: {
            flows: 12345,
            destination_ips: 12,
            source_ips: 55,
            country: 'US',
          },
          destination: null,
          network: {
            bytes_in: 3826633497,
            bytes_out: 1083495734,
          },
        },
        cursor: {
          value: '9.9.9.9',
        },
      },
    ],
    pageInfo: {
      activePage: 1,
      fakeTotalCount: 50,
      showMorePagesIndicator: true,
    },
  },
};
