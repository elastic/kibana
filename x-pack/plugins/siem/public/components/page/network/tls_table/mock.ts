/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { TlsData } from '../../../../graphql/types';

export const mockTlsData: TlsData = {
  totalCount: 2,
  edges: [
    {
      node: {
        _id: '2fe3bdf168af35b9e0ce5dc583bab007c40d47de',
        alternativeNames: ['*.elastic.co', 'elastic.co'],
        commonNames: ['*.elastic.co'],
        ja3: ['7851693188210d3b271aa1713d8c68c2', 'fb4726d465c5f28b84cd6d14cedd13a7'],
        issuerNames: ['DigiCert SHA2 Secure Server CA'],
        notAfter: ['2021-04-22T12:00:00.000Z'],
      },
      cursor: {
        value: '2fe3bdf168af35b9e0ce5dc583bab007c40d47de',
      },
    },
    {
      node: {
        _id: '61749734b3246f1584029deb4f5276c64da00ada',
        alternativeNames: ['api.snapcraft.io'],
        commonNames: ['api.snapcraft.io'],
        ja3: ['839868ad711dc55bde0d37a87f14740d'],
        issuerNames: ['DigiCert SHA2 Secure Server CA'],
        notAfter: ['2019-05-22T12:00:00.000Z'],
      },
      cursor: {
        value: '61749734b3246f1584029deb4f5276c64da00ada',
      },
    },
    {
      node: {
        _id: '6560d3b7dd001c989b85962fa64beb778cdae47a',
        alternativeNames: ['changelogs.ubuntu.com', 'manpages.ubuntu.com'],
        commonNames: ['changelogs.ubuntu.com'],
        ja3: ['da12c94da8021bbaf502907ad086e7bc'],
        issuerNames: ["Let's Encrypt Authority X3"],
        notAfter: ['2019-06-27T01:09:59.000Z'],
      },
      cursor: {
        value: '6560d3b7dd001c989b85962fa64beb778cdae47a',
      },
    },
  ],
  pageInfo: {
    endCursor: {
      value: '10',
    },
    hasNextPage: false,
  },
};
