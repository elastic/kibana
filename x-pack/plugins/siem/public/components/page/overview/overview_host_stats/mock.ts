/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { OverviewHostData } from '../../../../graphql/types';

export const mockData: { OverviewHost: OverviewHostData } = {
  OverviewHost: {
    auditbeatAuditd: 73847,
    auditbeatFIM: 107307,
    auditbeatLogin: 60015,
    auditbeatPackage: 2003,
    auditbeatProcess: 1200,
    auditbeatUser: 1979,
  },
};
