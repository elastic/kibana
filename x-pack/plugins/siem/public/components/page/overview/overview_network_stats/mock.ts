/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { OverviewNetworkData } from '../../../../graphql/types';

export const mockData: { OverviewNetwork: OverviewNetworkData } = {
  OverviewNetwork: {
    packetbeatFlow: 16,
    packetbeatDNS: 10277307,
    filebeatSuricata: 60015,
    filebeatZeek: 2003,
    auditbeatSocket: 12,
  },
};
