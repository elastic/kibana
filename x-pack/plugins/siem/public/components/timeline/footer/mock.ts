/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { EventsTimelineData } from '../../../graphql/types';

export const mockData: { Events: EventsTimelineData } = {
  Events: {
    totalCount: 15546,
    pageInfo: {
      hasNextPage: true,
      endCursor: {
        value: '1546878704036',
        tiebreaker: '10624',
      },
    },
    edges: [
      {
        cursor: {
          value: '1546878704036',
          tiebreaker: '10656',
        },
        node: {
          _id: 'Fo8nKWgBiyhPd5Zo3cib',
          timestamp: '2019-01-07T16:31:44.036Z',
          _index: 'auditbeat-7.0.0-2019.01.07',
          destination: {
            ip: ['24.168.54.169'],
            port: [62123],
          },
          event: {
            category: null,
            id: null,
            module: ['system'],
            severity: null,
            type: null,
          },
          geo: null,
          host: {
            name: ['siem-general'],
            ip: null,
          },
          source: {
            ip: ['10.142.0.6'],
            port: [9200],
          },
          suricata: null,
        },
      },
      {
        cursor: {
          value: '1546878704036',
          tiebreaker: '10624',
        },
        node: {
          _id: 'F48nKWgBiyhPd5Zo3cib',
          timestamp: '2019-01-07T16:31:44.036Z',
          _index: 'auditbeat-7.0.0-2019.01.07',
          destination: {
            ip: ['24.168.54.169'],
            port: [62145],
          },
          event: {
            category: null,
            id: null,
            module: ['system'],
            severity: null,
            type: null,
          },
          geo: null,
          host: {
            name: ['siem-general'],
            ip: null,
          },
          source: {
            ip: ['10.142.0.6'],
            port: [9200],
          },
          suricata: null,
        },
      },
    ],
  },
};
