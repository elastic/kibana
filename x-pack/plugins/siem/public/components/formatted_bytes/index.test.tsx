/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';

import { mockFrameworks, TestProviders } from '../../mock';

import { PreferenceFormattedBytes } from '.';

describe('formatted_bytes', () => {
  describe('PreferenceFormattedBytes', () => {
    describe('rendering', () => {
      const bytes = '2806422';

      test('renders correctly against snapshot', () => {
        const wrapper = shallow(<PreferenceFormattedBytes value={bytes} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });

      test('it renders bytes to hardcoded format when no configuration exists', () => {
        const wrapper = mount(
          <TestProviders mockFramework={{}}>
            <PreferenceFormattedBytes value={bytes} />
          </TestProviders>
        );
        expect(wrapper.text()).toEqual('2.676MB');
      });

      test('it renders bytes according to the default format', () => {
        const wrapper = mount(
          <TestProviders mockFramework={mockFrameworks.default_browser}>
            <PreferenceFormattedBytes value={bytes} />
          </TestProviders>
        );
        expect(wrapper.text()).toEqual('2.676MB');
      });

      test('it renders bytes supplied as a number according to the default format', () => {
        const wrapper = mount(
          <TestProviders mockFramework={mockFrameworks.default_browser}>
            <PreferenceFormattedBytes value={+bytes} />
          </TestProviders>
        );
        expect(wrapper.text()).toEqual('2.676MB');
      });

      test('it renders bytes according to new format', () => {
        const wrapper = mount(
          <TestProviders mockFramework={mockFrameworks.bytes_short}>
            <PreferenceFormattedBytes value={bytes} />
          </TestProviders>
        );
        expect(wrapper.text()).toEqual('3MB');
      });
    });
  });
});
