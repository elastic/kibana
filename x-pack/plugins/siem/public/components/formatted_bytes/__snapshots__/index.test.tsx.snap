// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferenceFormattedBytes renders correctly against snapshot 1`] = `
<Fragment>
  2.7MB
</Fragment>
`;
